src/pages/Home.js

```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+import logo from '../CodeKingTrans.png';
+import Navbar from '../components/Navbar';
+import Footer from '../components/Footer';
+import '../App.css';
+
+
+function Login() {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  useEffect(() => {
+    console.log("updated...")
+  }, [email, password]);
+
+  const login = useCallback(() => {
+    console.log(email);
+    console.log(password);
+    navigate('/user');
+  }, [email, password, navigate]);
```

Comment:
`useCallback` isn't really useful here.
 you want to memoize functions so they don't get recreated every render. In this case, the only things that could cause `Login` to rerender are if `email`, `password`, or `navigate` updates. And you have all three of those in your dependency array for the `useCallback` here, which functionally causes `login` to get recreated on every render of the `Login` component.

Reply:
If the `Login` component had more states that the `login` function didn't care about, then it would make sense to memoize the `login` function.